packages/azure-http-specs/specs/azure/resource-manager/resources/mockapi.ts

```diff
@@ -60,6 +61,139 @@ const validSingletonResource = {
+  id: `/subscriptions/${SUBSCRIPTION_ID_EXPECTED}/resourceGroups/${RESOURCE_GROUP_EXPECTED}/providers/Azure.ResourceManager.Resources/locations/${LOCATION_EXPECTED}/locationResources/resource`,
+  name: "resource",
+  type: "Azure.ResourceManager.Resources/locationResources",
+  properties: {
+    description: "valid",
+    provisioningState: "Succeeded",
+  },
+  systemData: {
+    createdBy: "AzureSDK",
+    createdByType: "User",
+    createdAt: "2024-10-04T00:56:07.442Z",
+    lastModifiedBy: "AzureSDK",
+    lastModifiedAt: "2024-10-04T00:56:07.442Z",
+    lastModifiedByType: "User",
+  },
+};
+
+// location resource
+Scenarios.Azure_ResourceManager_Resources_LocationResources_get = passOnSuccess({
+  uri: "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Resources/locations/:location/locationResources/:locationResourceName",
+  method: "get",
+  request: {
+    params: {
+      subscriptionId: SUBSCRIPTION_ID_EXPECTED,
+      resourceGroup: RESOURCE_GROUP_EXPECTED,
+      location: LOCATION_EXPECTED,
+      locationResourceName: "resource",
+      "api-version": "2023-12-01-preview",
+    },
+  },
+  response: {
+    status: 200,
+    body: json(validResourceGroupLocationResource),
+  },
+  kind: "MockApiDefinition",
+});
+
+Scenarios.Azure_ResourceManager_Resources_LocationResources_createOrUpdate = passOnSuccess({
+  uri: "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Resources/locations/:location/locationResources/:locationResourceName",
+  method: "put",
+  request: {
+    params: {
+      subscriptionId: SUBSCRIPTION_ID_EXPECTED,
+      resourceGroup: RESOURCE_GROUP_EXPECTED,
+      location: LOCATION_EXPECTED,
+      locationResourceName: "resource",
+      "api-version": "2023-12-01-preview",
+    },
+    body: {
+      properties: {
+        description: "valid",
+      },
+    },
+  },
+  response: {
+    status: 200,
+    body: json(validResourceGroupLocationResource),
+  },
+  kind: "MockApiDefinition",
+});
+
+Scenarios.Azure_ResourceManager_Resources_LocationResources_update = passOnSuccess({
+  uri: "/subscriptions/:subscriptionId/resourceGroups/:resourceGroup/providers/Azure.ResourceManager.Resources/locations/:location/locationResources/:locationResourceName",
+  method: "patch",
+  request: {
+    params: {
+      subscriptionId: SUBSCRIPTION_ID_EXPECTED,
+      resourceGroup: RESOURCE_GROUP_EXPECTED,
+      location: LOCATION_EXPECTED,
+      locationResourceName: "resource",
+      "api-version": "2023-12-01-preview",
+    },
+    body: {
+      properties: {
+        description: "valid2",
+      },
+    },
+    headers: {
+      "Content-Type": "application/json",
+    },
+  },
+  response: {
+    status: 200,
+    body: json({
+      ...validResourceGroupLocationResource,
+      properties: {
+        provisioningState: "Succeeded",
+        description: "valid2",
+      },
+    }),
+  },
+  kind: "MockApiDefinition",
+});
```

Comment:
Have you tested the PATCH via Java SDK? Does it need a handler?

Reply:
I have tested PATCH with the Java SDK, and it dose not need a handler.